OpenOversight/app/main/views.py

```diff
@@ -74,26 +80,49 @@ def submit_complaint():
                            officer_image=request.args.get('officer_image'))
 
 
-@main.route('/submit')
+@main.route('/submit', methods=['GET', 'POST'])
+@login_required
 def submit_data():
-    return render_template('submit.html')
-
-
-@main.route('/upload', methods=['POST'])
-def upload_file():
-    file = request.files['file']
-    if file and allowed_file(file.filename):
-        filename = secure_filename(file.filename)
-        file.save(os.path.join(current_app.config['UNLABELLED_UPLOADS'], filename))
-        return redirect(url_for('main.show_upload',
-                                filename=filename))
-
-
-@main.route('/show_upload/<filename>')
-def show_upload(filename):
-    # return send_from_directory('../'+config.UNLABELLED_UPLOADS,
-    #                           filename)
-    return 'Successfully uploaded: {}'.format(filename)
+    form = HumintContribution()
+    if request.method == 'POST' and form.validate_on_submit():
+        original_filename = secure_filename(request.files[form.photo.name].filename)
+        image_data = request.files[form.photo.name].read()
+
+        # See if there is a matching photo already in the db
+        hash_img = compute_hash(image_data)
+        hash_found = Image.query.filter_by(hash_img=hash_img).first()
+
+        if not hash_found:
+            # Generate new filename
+            file_extension = original_filename.split('.')[-1]
```

Comment:
Yep this is true, I've added a couple of tests in 5548ba5 to check that files named "passwd" (due to the way that WTForms validates the file upload form, the file has to exist, so I did this "passwd" check in order to get that test passing both on the dev VM and on Travis) and files with invalid file extensions are not uploaded.